### ISSUE TITLE
fill: transparent; produced by Vega is not compliant with SVG spec

### ISSUE THREAD

**Issue by alshain** [vega/vega#1080]
Vega produces SVG output that is not spec-compliant.

In particular, it produces `fill: transparent`, where `transparent` appears to be invalid per the spec. [1](https://www.w3.org/TR/SVG11/painting.html#FillProperties), [2](https://www.w3.org/TR/SVG11/painting.html#SpecifyingPaint), it's also not included in the list of [valid CSS2 color keywords](https://www.w3.org/TR/SVG11/types.html#ColorKeywords)

This is a problem because: Other software such as Inkscape/IntelliJ fall back to black paint when the invalid `transparent` is encountered. This is a problem when trying to [convert the SVG output to Latex as suggested here.](https://tex.stackexchange.com/questions/2099/how-to-include-svg-diagrams-in-latex/2107#2107).

**Possible fix**:

[Use `fill-opacity: 0` instead<sup>spec</sup>](https://www.w3.org/TR/SVG11/painting.html#FillOpacityProperty)

**Reproduce**

Try the example in the new editor: https://vega.github.io/editor/
Set the output to SVG

The error manifests in the first `<path>` element.

`<path class="background" d="M0.5,0.5h200v200h-200Z" style="fill: transparent; stroke: rgb(221, 221, 221);"></path>`

**Example**
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v1.2.1.json",
  "data": {
    "values": [
      {
        "number_of_clients": 1,
        "response_time": 56,
        "throughput": 10
      },
      {
        "number_of_clients": 1,
        "response_time": 50,
        "throughput": 15
      },
      {
        "number_of_clients": 1,
        "response_time": 43,
        "throughput": 12
      },
      {
        "number_of_clients": 2,
        "response_time": 50,
        "throughput": 10
      },
      {
        "number_of_clients": 2,
        "response_time": 65,
        "throughput": 30
      }
    ]
  },
  "encoding": {
    "x": {
      "field": "number_of_clients"
    },
    "y": {
      "aggregate": "average",
      "field": "throughput"
    }
  },
  "mark": "bar"
}
```

Links:
https://bugs.launchpad.net/inkscape/+bug/1735012
https://youtrack.jetbrains.com/issue/IDEA-182816

**Comment by alshain** [comment]
The spec entry I linked doesn't directly apply, because it's not a property on the path element, but rather inside the `style` property.

However, [in the section about styling and inline styles](https://www.w3.org/TR/SVG11/styling.html#StyleAttribute), it says



> When CSS styling is used, CSS inline style is specified by including semicolon-separated property declarations of the form "name : value" within the ‘style’ attribute. Property declarations must follow CSS style rules thus CSS defined properties (e.g. 'font-size') when having a <length> value must include a unit (for non-zero values). See SVG's styling properties for a list of CSS defined properties.

Linked:

> The following SVG properties are not defined in CSS2. The complete normative definitions for these properties are found in this specification:
> *...*
> ‘fill’


Which links back to [the spec entry](https://www.w3.org/TR/SVG11/painting.html#FillProperty) I linked above.

**Comment by danyelf** [comment]
This is mostly a +1, and I'm providing an example:

* An SVG file. Note that the background is set to white.
* The way that PDF renders it.

For now, I'm actually going through and post-processing the files with a text editor

[singleview_5_13a_clustered-bar-chart.zip](https://github.com/vega/vega/files/1560995/singleview_5_13a_clustered-bar-chart.zip)

**Comment by jheer** [comment]
Thanks @alshain for raising this issue. Indeed, we would like output SVG to be v1.1 compliant!

There are currently two cases where Vega internally applies a "transparent" fill type (set via the parser config): as the default fill for plotting symbols within legends and as the background color for "cell" group marks created by Vega-Lite. In other cases, the default background fill color for both the full visualization and for group marks is `none`, which is a legal SVG value.

I will work on changing the legend plotting symbol defaults now. For the "cell" type, I will need to coordinate with the lead Vega-Lite devs before pushing a change that potentially breaks other aspects (such as input event capture in the browser). The example from @danyelf suffers from this Vega-Lite `cell` styling issue.

In the meantime, one potential fix is to [change the configuration settings](https://vega.github.io/vega/docs/config/), either by providing an external configuration file or providing configuration directives directly within a Vega spec. Here is an example of the latter:

```json
"config": {
  "legend": { "symbolFillColor": "none" },
  "style": {
    "cell": { "fill": "none" }
  }
}
```

Paging @arvind, @domoritz, @kanitw. We need to change the default settings for the `cell` style used by Vega-Lite. One option might be to use `{fill: #fff, fill-opacity: 0}`, as this would still capture mouse events. However, this might affect other aspects. Please advise. Also note that if the Vega-Lite compiler generates `fill: transparent` elsewhere, this will result in illegal output SVG upon export.

**Comment by jheer** [comment]
Fixed in vega-scenegraph v2.1.0 and released in vega 3.0.9. Any `transparent` color values are now mapped to `none` when rendered by the SVGStringRenderer.
